api/src/auth/drivers/saml.ts

```diff
@@ -95,14 +95,8 @@ export class SAMLAuthDriver extends LocalAuthDriver {
 	}
 
 	async login(user: User): Promise<void> {
-		let authData = user.auth_data as AuthData;
-
-		if (typeof authData === 'string') {
-			try {
-				authData = parseJSON(authData);
-			} catch {
-				logger.warn(`[SAML] Session data isn't valid JSON: ${authData}`);
-			}
+		if (!user.auth_data) {
```

Comment:
What is stored in auth data? Seeing we're not using it for anything in login, why is it required that it's set to something? 🤔

Reply:
mostly this was inherited as we use it so I am currently removing it, mostly just used it to check.

Reply:
This throwing of error broke the tests. 🤔

Reply:
I'm not sure we can remove it since it'll inherit the login from the local provider which expects a username and password. It should just be an empty method here, since we've already handled user verification.

Reply:
> I'm not sure we can remove it since it'll inherit the login from the local provider which expects a username and password. It should just be an empty method here, since we've already handled user verification.

yeah just put it back as an empty method with a return.

Reply:
We also have a bit of weirdness in the SSO flows to keep compatibility with some flows like TFA which require a user to enter a password.

Technically an SSO user can have a password (manually set after they are registered) which is required to activate TFA. So the existing SSO providers inherit the Local auth driver so that the `verify` function is correctly triggered for enabling TFA.

There was talk about improving the TFA flow for SSO users, but I'm not sure it was ever implemented.

Reply:
will re-re-resolve this once I make sure putting the empty method back in fixed @licitdev's tests.

Reply:
> There was talk about improving the TFA flow for SSO users, but I'm not sure it was ever implemented.

Ref: https://github.com/directus/directus/issues/12562

Reply:
tests fixed